sway-core/src/asm_generation/from_ir.rs

```diff
@@ -1334,7 +1335,7 @@ impl<'ir> AsmBuilder<'ir> {
     fn compile_read_register(&mut self, instr_val: &Value, reg: &sway_ir::Register) {
         let instr_reg = self.reg_seqr.next();
         self.bytecode.push(Op {
-            opcode: Either::Left(VirtualOp::LW(
+            opcode: Either::Left(VirtualOp::MOVE(
```

Comment:
Hmmm. Shouldn't you still keep the logic to `LW` a _constant_ into the register, if a constant is used for the gas to forward rather than default?

Reply:
So that's actually handled here: https://github.com/FuelLabs/sway/blob/85032fb96c55efa9fbe3ac00eb5b3b0c5c2af051/sway-core/src/optimize.rs#L873

The `read_register` instruction that we're handling here is only meant to extract the value of the constant register when needed, and that's only needed in the default case for `gas`.